ocp_qp/x_ocp_qp.c

```diff
@@ -483,6 +483,206 @@ void CVT_ROWMAJ_TO_OCP_QP(REAL **A, REAL **B, REAL **b, REAL **Q, REAL **S, REAL
 	}
 
 
+// convert generic field
+void CVT_COLMAJ_TO_OCP_QP_GF(char *field_name, int stage, REAL *in, struct OCP_QP *qp)
+	{
+	// extract dim
+	int *nx = qp->dim->nx;
+	int *nu = qp->dim->nu;
+	int *nb = qp->dim->nb;
+	int *ng = qp->dim->ng;
+	int *ns = qp->dim->ns;
+    
+	// matrices
+	if(hpipm_strcmp(field_name, "A")) 
+		CVT_TRAN_MAT2STRMAT(nx[stage+1], nx[stage], in, nx[stage+1], qp->BAbt+stage, nu[stage], 0);
+	else if(hpipm_strcmp(field_name, "B")) 
+		CVT_TRAN_MAT2STRMAT(nx[stage+1], nu[stage], in, nx[stage+1], qp->BAbt+stage, 0, 0);
+	else if(hpipm_strcmp(field_name, "Q")) 
+		CVT_MAT2STRMAT(nx[stage], nx[stage], in, nx[stage], qp->RSQrq+stage, nu[stage], nu[stage]);
+	else if(hpipm_strcmp(field_name, "S")) 
+		CVT_TRAN_MAT2STRMAT(nu[stage], nx[stage], in, nu[stage], qp->RSQrq+stage, nu[stage], 0);
+	else if(hpipm_strcmp(field_name, "R")) 
+		CVT_MAT2STRMAT(nu[stage], nu[stage], in, nu[stage], qp->RSQrq+stage, 0, 0);
+	else if(hpipm_strcmp(field_name, "C")) 
+		CVT_TRAN_MAT2STRMAT(ng[stage], nx[stage], in, ng[stage], qp->DCt+stage, nu[stage], 0);
+	else if(hpipm_strcmp(field_name, "D")) 
+		CVT_TRAN_MAT2STRMAT(ng[stage], nu[stage], in, ng[stage], qp->DCt+stage, 0, 0);
+	// vectors
+	else if(hpipm_strcmp(field_name, "b") == 1) { 
+		int row_offset = qp->dim->nx[stage] + qp->dim->nu[stage], col_offset = 0;
+		CVT_TRAN_MAT2STRMAT(nx[stage+1], 1, in, nx[stage+1], &(qp->BAbt[stage]), row_offset, col_offset);
```

Comment:
probably this are not necessary any longer (that was the aim at least, I should double-check), if this is the case also the other setters should be cleaned one day :p